### PR TITLE
multiple improvments (liveness/readiness Probes, ingress fix, rollingUpdate)

### LINKED ISSUE #98: Provide helm logic for additional resource labels
Hello,
When deploying litmus resources to my environment, there are a few changes I need to make to meet infrastructure standards.  Among them are identifying labels on resources to help correlate them with usage and owner.  Following our discussion at https://kubernetes.slack.com/archives/CNXNB0ZTN/p1616624243043700 , @Jasstkn suggested that the helm charts could be refactored to render labels from values I provide, if they exist.  This would be very helpful, let's please discuss.

### PR DESCRIPTION
**Pull request by Vr00mm**
Support for limit / request resource values ​​in mongo and server (authServer) templates

Signed-off-by: Rémi ZIOLKOWSKI <remi.ziolkowski-ext@pole-emploi.fr>

#### What this PR does / why we need it:
- This PR permit to set or limit ressources for authServer ( container of server-deployment.yml ) and for mongo ( container of mongo-sts.yml ) 
- This PR permit to override resources name 
- This PR permit to set custom labels per components ( Hi @immanetize )
- This PR fix ingress on the default path (/ -> frontend)
- This PR define liveness and readiness for mongo, frontend and graphql-server
( ill PR later the litmus-portal to add a "status" route into authn-server -> https://github.com/litmuschaos/litmus/blob/master/litmus-portal/authentication/api/routes/user_router.go#L13 and then we will can get a readiness on auth-server)

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #98

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

### CODE REVIEW

File: charts/litmus-2-0-0-beta/templates/mongo-sts.yaml [hunk 1]
```diff
@@ -34,6 +34,8 @@ spec:
             - containerPort: {{ .Values.mongo.containerPort }}
               name: http
           imagePullPolicy: {{ .Values.mongo.image.pullPolicy }}
+          resources:
```
Comment (rajdas98) [inline charts/litmus-2-0-0-beta/templates/mongo-sts.yaml:37]: Can you add an if condition for checking non-empty resources?

File: charts/litmus-2-0-0-beta/templates/server-deployment.yaml [hunk 2]
```diff
@@ -95,6 +95,8 @@ spec:
           ports:
             - containerPort: {{ .Values.portal.server.authServer.containerPort }}
               name: auth-server
+          resources:
```
Comment (rajdas98) [inline charts/litmus-2-0-0-beta/templates/server-deployment.yaml:98]: same here

**Review by rajdas98 — CHANGES_REQUESTED** [review]

File: charts/litmus-2-0-0-beta/templates/mongo-sts.yaml (hunk 1 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/mongo-sts.yaml:37]: Hello Raj, Yes i can. But we need it ?
https://github.com/litmuschaos/litmus-helm/blob/master/charts/litmus-2-0-0-beta/values.yaml#L45

Or I should add the same condition for the frontend and graphServer templates ?

Comment (rajdas98) [reply charts/litmus-2-0-0-beta/templates/mongo-sts.yaml:37]: >Hello Raj, Yes i can. But we need it ?
https://github.com/litmuschaos/litmus-helm/blob/master/charts/litmus-2-0-0-beta/values.yaml#L45

Okay, then it's not required

> Or I should add the same condition for the frontend and graphServer templates ?

Yes, please.

BTW thanks for this contribution.

**Review by rajdas98 — APPROVED** [review]

**Comment by rajdas98** [comment]
@Vr00mm circleci build is failed.

**Comment by Vr00mm** [comment]
> @Vr00mm circleci build is failed.

Iam not in :)
"ERROR: failed to create cluster: failed to generate kubeadm config content: failed to get kubernetes version from node: failed to get file: command "docker exec --privileged kind-control-plane cat /kind/version" failed with error: exit status 126"

**Comment by Vr00mm** [comment]
I writted some more code for {live,readi}-nessProbes and  Strategy.rollingUpdate
Ill push another PR tomorow.

See you later,
Best regards

**Comment by Vr00mm** [comment]
Hello, can @ispeakc0de or @Jasstkn can review this pr please ?

File: charts/litmus-2-0-0-beta/README.md [hunk 3]
```diff
@@ -51,26 +51,50 @@ $ helm install litmus-portal litmuschaos/litmus-2-0-0-beta
...
+| portal.frontend.livenessProbe.periodSeconds | int | `10` |  |
+| portal.frontend.livenessProbe.successThreshold | int | `1` |  |
+| portal.frontend.livenessProbe.timeoutSeconds | int | `5` |  |
+| portal.frontend.readinessProbe.initialDelaySeconds | int | `5` |  |
+| portal.frontend.readinessProbe.periodSeconds | int | `10` |  |
+| portal.frontend.readinessProbe.successThreshold | int | `1` |  |
+| portal.frontend.readinessProbe.timeoutSeconds | int | `1` |  |
 | portal.frontend.replicas | int | `1` |  |
 | portal.frontend.resources | object | `{}` |  |
+| portal.frontend.rollingUpdate.enabled | bool | `false` |  |
```
Comment (Jasstkn) [inline charts/litmus-2-0-0-beta/README.md:95]: why default is false? 

It's common to allow users to decide which strategy they want to use, default: RollingUpdate
```
{{- if .Values.frontend.updateStrategy }}
{{ toYaml .Values.frontend.updateStrategy | indent 2 }}
{{- end }}

  ## Strategy for deployment updates.
  ##
  ## Example:
  ##
  ##   strategy:
  ##     type: RollingUpdate
  ##     rollingUpdate:
  ##       maxSurge: 1
  ##       maxUnavailable: 25%
  strategy: {}
```

File: charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml [hunk 4]
```diff
@@ -25,8 +38,20 @@ spec:
         - name: litmusportal-frontend
           image: {{ .Values.image.imageRegistryName }}/{{ .Values.portal.frontend.image.repository }}:{{ .Values.portal.frontend.image.tag }}
           imagePullPolicy: {{ .Values.portal.frontend.image.pullPolicy }}
+          {{- if .Values.portal.frontend.resources }}
```
Comment (Jasstkn) [inline charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml:41]: you can replace it with:
```
          resources:
            {{- toYaml .Values.resources | nindent 12 }}
```

File: charts/litmus-2-0-0-beta/templates/mongo-sts.yaml [hunk 5]
```diff
@@ -34,6 +40,26 @@ spec:
             - containerPort: {{ .Values.mongo.containerPort }}
               name: http
           imagePullPolicy: {{ .Values.mongo.image.pullPolicy }}
+          {{- if .Values.mongo.resources }}
+          resources:
+            {{- toYaml .Values.mongo.resources | nindent 12 }}
```
Comment (Jasstkn) [inline charts/litmus-2-0-0-beta/templates/mongo-sts.yaml:45]: you can replace it with:
```
          resources:
            {{- toYaml .Values.resources | nindent 12 }}
```

**Review by Jasstkn — CHANGES_REQUESTED** [review]
Hi! I've asked a couple of questions.

File: charts/litmus-2-0-0-beta/README.md (hunk 3 above)
Comment (HexalGJ) [reply charts/litmus-2-0-0-beta/README.md:95]: Hello, I set this default value to false so as not to impact the current behavior of a deployment already done.
I will make the modification in this way

File: charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml (hunk 4 above)
Comment (HexalGJ) [reply charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml:41]: Rajdas98 asked for [this change:](https://github.com/litmuschaos/litmus-helm/pull/142/files/94ae404ea82902c5011dd0577589cd43eacd7988)
The idea is to not generate 'empty' resources line in the manifest i think

File: charts/litmus-2-0-0-beta/templates/mongo-sts.yaml (hunk 5 above)
Comment (HexalGJ) [reply charts/litmus-2-0-0-beta/templates/mongo-sts.yaml:45]: same

File: charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml (hunk 4 above)
Comment (Jasstkn) [reply charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml:41]: @rajdas98 can you explain your point please? those conditions are ugly to read and I don't see if they are really applicable here.

File: charts/litmus-2-0-0-beta/README.md (hunk 3 above)
Comment (HexalGJ) [reply charts/litmus-2-0-0-beta/README.md]: I implemented it in the same way than resources ( with a value check).
according to @rajdas98 answer, I'll make needed edits for resources and strategy

File: charts/litmus-2-0-0-beta/templates/server-deployment.yaml [hunk 6]
```diff
@@ -29,8 +39,20 @@ spec:
...
             - containerPort: {{ .Values.portal.server.graphqlServer.containerPort }}
               name: graphql-srver
+          {{- if .Values.portal.server.graphqlServer.resources }}
           resources:
             {{- toYaml .Values.portal.server.graphqlServer.resources | nindent 12 }}
+          {{- end }}
+          livenessProbe:
+            httpGet:
+              path: /
+              port: graphql-srver
```
Comment (rajdas98) [inline charts/litmus-2-0-0-beta/templates/server-deployment.yaml:49]:
```suggestion
              port: graphql-server
```

File: charts/litmus-2-0-0-beta/templates/ingress.yaml [hunk 7]
```diff
@@ -53,7 +53,7 @@ spec:
               number: 9002
     {{- else if semverCompare "<1.19-0" .Capabilities.KubeVersion.GitVersion }}
       paths:
-      - path: /(.*)
+      - path: /
```
Comment (rajdas98) [inline charts/litmus-2-0-0-beta/templates/ingress.yaml:56]: I think before 1.19, we need to add (.*)

File: charts/litmus-2-0-0-beta/templates/server-deployment.yaml (hunk 6 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/server-deployment.yaml:49]: Hello Raj, thats the name of the port, i can edit it in the deployment template too if you want.

File: charts/litmus-2-0-0-beta/templates/ingress.yaml (hunk 7 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/ingress.yaml:56]: I think its a Path Type configuration, not a kubernetes capabilities version ?
There is nothing about (.*) in the 1.18 documentation:
https://v1-18.docs.kubernetes.io/docs/concepts/services-networking/ingress/

We can leave it empty to set the default service to frontend.

I go the error thats why i fixed it.
Deployed with /(.*)
i cannot access https://my-domain-with-litmus.com/
i can access to https://my-domain-with-litmus.com/login

Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/ingress.yaml:56]: I understood the issue.

https://github.com/litmuschaos/litmus-helm/blob/master/charts/litmus-2-0-0-beta/values.yaml#L30
Here the annotation is a commentary and there is nothing about the requirement of this.

This annotation is only valid for ingress nginx.
For traefik, the annotation should be :
`traefik.ingress.kubernetes.io/rewrite-target: $1
`
What about put those values as default ?
`
`nginx.ingress.kubernetes.io/rewrite-target: /$1
traefik.ingress.kubernetes.io/rewrite-target: $1
`

I remove the / cause it generate dirty http request on '/(.*)' like -> //statics/icons/...

**Comment by Vr00mm** [comment]
Hello guys,

I push the readiness/liveness for graphql tomorow
with the asked changes.

See you later,
Best regards

File: charts/litmus-2-0-0-beta/templates/server-deployment.yaml (hunk 6 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/server-deployment.yaml]: done

File: charts/litmus-2-0-0-beta/templates/ingress.yaml (hunk 7 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/ingress.yaml]: Reverted, done

File: charts/litmus-2-0-0-beta/templates/mongo-sts.yaml (hunk 5 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/mongo-sts.yaml:45]: done

File: charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml (hunk 4 above)
Comment (Vr00mm) [reply charts/litmus-2-0-0-beta/templates/frontend-deployment.yaml]: done

**Comment by rajdas98** [comment]
Hi @Vr00mm There are some merge conflicts. Can you resolve it?

**Comment by Vr00mm** [comment]
> Hi @Vr00mm There are some merge conflicts. Can you resolve it?

Hello @rajdas98 it should be OK for merge conflicts.

**Comment by Vr00mm** [comment]
Ready to merge for me

**Review by Jasstkn — APPROVED** [review]
LGTM

**Comment by Vr00mm** [comment]
Thanx for the merge

**Pull request merged**